src/App.tsx

```diff
@@ -1,93 +1,224 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
+
+import {
+  getTodos,
+  updateTodo,
+  createTodo,
+  deleteTodo,
+} from './api/todos';
+import { AuthContext } from './components/Auth/Auth.Context';
+import { Footer } from './components/Footer/Footer';
+import { TodoList } from './components/Todolist/TodoList';
+import { Todo, SelectedType } from './types/Todo';
+import { TodoTitle } from './types/TodoTitle';
+import { ErrorType } from './components/ErrorType/ErrorType';
+import { TodoCompleted } from './types/TodoCompleted';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState<string>(SelectedType.ALL);
+  const [showAll, setShowAll] = useState(true);
+  const [deletingloader, setDeletingLoader] = useState<number[]>([]);
+  const [changeAllTodos, setChangeAllTodos] = useState<number[]>([]);
+  const [errorType, setErrorType] = useState('');
+  const user = useContext(AuthContext);
+  const [isAdding, setIsAdding] = useState(false);
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const [loaderVisibility, setIsLoaderVisibility] = useState<number>(0);
+
+  const handleError = (text: string) => {
+    setErrorType(text);
+    setTimeout(() => {
+      setErrorType('');
+    }, 5000);
+  };
+
+  const handleNewTodos = async () => {
+    if (!query.trim().length) {
```

Comment:
`!query.trim()` - will be enough, because an empty string in javascript means `false`

```suggestion
    if (!query.trim()) {
```